pkg/controller/cloudfront/distribution/lateinit.go

```diff
@@ -36,6 +36,11 @@ import (
 // what the AWS SDK says that some of these fields are required.
 
 func lateInitialize(dp *svcapitypes.DistributionParameters, gdo *svcsdk.GetDistributionOutput) error {
+	// during import scenarios, we need the original caller reference info
+	if gdo.Distribution != nil && gdo.Distribution.DistributionConfig != nil {
```

Comment:
Did you consider having an `Initializer` like the ones we use for [tagging](https://github.com/crossplane/provider-aws/blob/793e588/pkg/controller/dynamodb/table/hooks.go#L78)? It'd basically populate the caller reference with UID if it's not given and then we'd be able to treat that field just like other fields in all calls since managed reconciler makes sure the initializers are called [before](https://github.com/crossplane/crossplane-runtime/blob/86fb15d/pkg/reconciler/managed/reconciler.go#L704) any external operation. In fact, we could even remove its `ignore` statement [here](https://github.com/crossplane/provider-aws/blob/86b704e/apis/cloudfront/generator-config.yaml#L14).

Reply:
Hi @muvaf, I tried to use an `Initializer` which sets `spec.forProvider.distributionConfig.callerReference` value with `UID` if it's not given. But, the import scenario failed again. Setting the value in `preCreate` guarantees that the resource does not exist on the cloud provider. 

Thanks for pointing out the ignore statement. I missed that one.